lib/galaxy/tools/__init__.py

```diff
@@ -301,6 +301,22 @@ class safe_update(NamedTuple):
 }
 
 
+def get_safe_version(tool: "Tool", requested_tool_version: str) -> Optional[str]:
+    if tool.id:
+        safe_version = WORKFLOW_SAFE_TOOL_VERSION_UPDATES.get(tool.id)
+        if safe_version and tool.lineage:
+            # tool versions are sorted from old to new, so check newest version first
+            for lineage_version in reversed(tool.lineage.tool_versions):
+                if (
+                    safe_version.current_version >= parse_version(lineage_version) >= safe_version.min_version
+                    and safe_version.current_version
+                    >= parse_version(requested_tool_version)
+                    >= safe_version.min_version
```

Comment:
This could be taken out of the loop and moved with the other checks at line 307.

Reply:
Pushed that into the first commit